WearTilesKotlin/app/src/main/java/com/example/wear/tiles/messaging/MessagingTileLayout.kt

```diff
@@ -100,83 +96,77 @@ private fun searchLayout(
     .setButtonColors(ButtonColors.secondaryButtonColors(MessagingTileTheme.colors))
     .build()
 
-@Preview(device = WearDevices.SMALL_ROUND, fontScale = 1.24f)
-@Preview(device = WearDevices.LARGE_ROUND, fontScale = 0.94f)
-private fun MessageTilePreview(context: Context): TilePreviewData {
+@Preview(device = WearDevices.SMALL_ROUND)
+@Preview(device = WearDevices.LARGE_ROUND)
+private fun messagingTilePreview(context: Context): TilePreviewData {
     val state = MessagingTileState(MessagingRepo.knownContacts)
-    return TilePreviewData(onTileResourceRequest = resources {
-        addIdToImageMapping(
-            state.contacts[1].imageResourceId(),
-            bitmapToImageResource(
-                BitmapFactory.decodeResource(context.resources, R.drawable.ali)
```

Comment:
Using `drawableResToImageResource` seems like an improvement.

You could consider inlining it if you want to minimise horologist usage here.

```
public fun drawableResToImageResource(
    @DrawableRes id: Int,
): ImageResource =
    ImageResource.Builder()
        .setAndroidResourceByResId(
            AndroidImageResourceByResId.Builder()
                .setResourceId(id)
                .build(),
        )
        .build()
```

Reply:
I was considering an extension function to enable `addIdToImageMapping()` to take a DrawableRes _or_ ImageResource:

```kotlin
fun ResourceBuilders.Resources.Builder.addIdToImageMapping(
    id: String,
    @DrawableRes resId: Int
): ResourceBuilders.Resources.Builder = addIdToImageMapping(
    id, ResourceBuilders.ImageResource.Builder()
        .setAndroidResourceByResId(
            ResourceBuilders.AndroidImageResourceByResId.Builder()
                .setResourceId(resId)
                .build()
        )
        .build()
)
```

Not sure whether this is idiomatic (versus the function approach) or whether it should be called `addIdToResourceMapping()`, etc. so was leaving it for a future PR.

Reply:
Added enhanced `addIdToImageMapping()` in https://github.com/android/wear-os-samples/pull/1164.